test/custom_task_test.go

```diff
@@ -401,3 +402,59 @@ spec:
 		t.Fatalf("Failed to get PipelineRun `%s`: %s", pipelineRun.Name, err)
 	}
 }
+
+func applyController(t *testing.T) {
+	t.Log("Creating Wait Custom Task Controller...")
+	cmd := exec.Command("ko", "apply", "-f", "./config/controller.yaml")
+	cmd.Dir = "./wait-task"
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		t.Fatalf("Failed to create Wait Custom Task Controller: %s, Output: %s", err, out)
+	}
+}
+
+func cleanUpController(t *testing.T) {
+	t.Log("Tearing down Wait Custom Task Controller...")
+	cmd := exec.Command("ko", "delete", "-f", "./config/controller.yaml")
+	cmd.Dir = "./wait-task"
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		t.Fatalf("Failed to tear down Wait Custom Task Controller: %s, Output: %s", err, out)
+	}
+}
+
+func TestCustomTask_NoCRD(t *testing.T) {
+	ctx := context.Background()
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	c, namespace := setup(ctx, t, requireAnyGate(supportedFeatureGates))
+	knativetest.CleanupOnInterrupt(func() { tearDown(ctx, t, c, namespace) }, t.Logf)
+	defer tearDown(ctx, t, c, namespace)
+
+	// Create a custom task controller
+	applyController(t)
+	defer cleanUpController(t)
+
+	// metadataLabel := map[string]string{"test-label": "test"}
+	runName := helpers.ObjectNameForTest(t)
+	if _, err := c.RunClient.Create(
+		ctx,
+		parse.MustParseRun(t, fmt.Sprintf(`
+metadata:
+  name: %s
+spec:
+  ref:
+    apiVersion: %s
+    kind: %s
+  params:
+  - name: duration
+    value: 1s
+`, runName, apiVersion, kind)), metav1.CreateOptions{}); err != nil {
+		t.Fatalf("Failed to create TaskRun %q: %v", runName, err)
+	}
+
+	// Wait for the Run to Finish.
+	if err := WaitForRunState(ctx, c, runName, time.Minute, Succeed(runName), "RunFinished"); err != nil {
+		t.Fatalf("Waiting for Run to finish running: %v", err)
+	}
```

Comment:
could you please add a check that the `run` is what we expect it to be? that is, compare the resolved run with the expected run?

Reply:
Yeah SG! Done.